src/cli-sdk/src/pack-tarball.ts

```diff
@@ -16,181 +18,316 @@ export type PackTarballResult = {
   shasum?: string
 }
 
+/**
+ * Replace workspace: and catalog: specs with actual versions
+ * @param {Manifest} manifest_ - The manifest to process
+ * @param {LoadedConfig} config - The loaded configuration containing project root, monorepo, and catalog data
+ * @returns {Manifest} The manifest with replaced specs
+ */
+const replaceWorkspaceAndCatalogSpecs = (
+  manifest_: Manifest,
+  config: LoadedConfig,
+): Manifest => {
+  // Create a deep copy of the manifest to avoid modifying the original
+  const manifest = JSON.parse(JSON.stringify(manifest_)) as Manifest
```

Comment:
I think structuredClone will be better to use here, wdyt?

```suggestion
  const manifest = structuredClone(manifest_) as Manifest
```